### PR TITLE
feat(wallet-connect): make v2 the default walletconnect method

### PR DESCRIPTION
**Pull request by JFrankfurt**
## Description

Makes version 2 of WalletConnect the default connection method. Version 1 is still accessible via the `...` menu toggle in the WalletModal.

This PR also includes a couple small fixes: 
1. `!important` on popover title text margin-bottom
2. removes the `...` menu when the option is pending such that it no longer is occluded by the `Loader` component.

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2231/retire-wallet-connect-10

### Before
|    Mobile    |   Desktop    |
| ------------ | ------------ |
| image| image|

### After
|    Mobile    |   Desktop   |
| ------------ | ----------- |
| image| image|

### QA (ie manual testing)

- [ ] Connect via wcv2 by clicking the main menu option
- [ ] Connect via wcv1 by clicking the `...` button on the WalletConnect option
- [ ] Confirm expected behavior per testing doc: https://www.notion.so/uniswaplabs/Uniswap-WalletConnect-v2-Test-Instructions-External-4b1176c29f9741cf8fcab77880b4878a?pvs=4

### CODE REVIEW

File: src/components/WalletModal/Option.tsx [hunk 1]
```diff
@@ -141,15 +141,15 @@ function PopupButtonContent({ connection, isDarkMode, show, onClick, onClose }:
...
+    <WCv1PopoverContent onClick={onClick} ref={popoverElement} size={ButtonSize.small} emphasis={ButtonEmphasis.medium}>
       <IconWrapper>
-        <WCv2Icon src={connection.getIcon?.(isDarkMode)} alt={connection.getName()} />
+        <WCv1Icon src={connection.getIcon?.(isDarkMode)} alt={connection.getName()} />
       </IconWrapper>
       <div>
-        <WCv2BodyText>Connect with v2</WCv2BodyText>
-        <WCv2Caption color="textSecondary">Under development and unsupported by most wallets</WCv2Caption>
+        <WCv1BodyText>Connect with v1</WCv1BodyText>
+        <WCv1Caption color="textSecondary">Support for v1 will be discontinued very soon.</WCv1Caption>
```
Comment (zzmp) [inline src/components/WalletModal/Option.tsx:150]: nit: I think "very" is unnecessary. CC @infredible

File: src/components/WalletModal/Option.tsx [hunk 2]
```diff
@@ -158,30 +158,32 @@ interface OptionProps {
...
     })
   }
-  const handleClickOpenWCv2Tooltip = (e: MouseEvent<HTMLButtonElement>) => {
+  const handleClickOpenWCv1Tooltip = (e: MouseEvent<HTMLButtonElement>) => {
     e.stopPropagation()
-    setWC2PromptOpen(true)
+    setWC1PromptOpen(true)
   }
 
+  const showExtraMenuToggle = connection.type === ConnectionType.WALLET_CONNECT_V2 && !isCurrentOptionPending
```
Comment (zzmp) [inline src/components/WalletModal/Option.tsx:185]: @JFrankfurt was this added as an extra check to fix a visual bug in the existing logic? lgtm, just wondering why it's the only addition here.

File: src/connection/WalletConnect.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ export class WalletConnectPopup extends WalletConnect {
 }
 
 // Custom class for Uniswap Wallet specific functionality
-export class UniwalletConnect extends WalletConnectPopup {
+export class UniwalletConnect extends WalletConnectV1WithAnalytics {
```
Comment (zzmp) [inline src/connection/WalletConnect.ts:34]: Why isn't this being migrated to v2? Should a TODO/ticket be added/created?

File: src/connection/WalletConnectV2.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ export class WalletConnectV2Popup extends WalletConnect {
         chains: [SupportedChainId.MAINNET],
         showQrModal: qrcode,
         rpcMap: RPC_URLS_WITHOUT_FALLBACKS,
+        optionalMethods: ['eth_signTypedData', 'eth_signTypedData_v4', 'eth_sign'],
```
Comment (zzmp) [inline src/connection/WalletConnectV2.ts:35]: Can you precede this line with a comment linking to the docs explaining why this is needed?

File: src/connection/WalletConnectV2.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ const RPC_URLS_WITHOUT_FALLBACKS = Object.entries(RPC_URLS).reduce(
 )
 const optionalChains = [...L1_CHAIN_IDS, ...L2_CHAIN_IDS].filter((x) => x !== SupportedChainId.MAINNET)
 
-export class WalletConnectV2Popup extends WalletConnect {
```
Comment (zzmp) [inline src/connection/WalletConnectV2.ts:19]: nit: just `WalletConnectV2` to match the filename. `Analytics` is an implementation detail.

File: src/connection/WalletConnect.ts [hunk 6]
```diff
@@ -14,7 +14,7 @@ const RPC_URLS_WITHOUT_FALLBACKS = Object.entries(RPC_URLS).reduce(
   {}
 )
 
-export class WalletConnectPopup extends WalletConnect {
```
Comment (zzmp) [inline src/connection/WalletConnect.ts:17]: nit: just `WalletConnectV1`. `Analytics` is an implementation detail.

File: src/connection/index.ts [hunk 7]
```diff
@@ -70,28 +70,28 @@ export const gnosisSafeConnection: Connection = {
...
 }
 
-const [web3WalletConnectV2, web3WalletConnectV2Hooks] = initializeConnector<WalletConnectV2Popup>(
-  (actions) => new WalletConnectV2Popup({ actions, onError })
+const [web3WalletConnectV2, web3WalletConnectV2Hooks] = initializeConnector<WalletConnectV2WithAnalytics>(
+  (actions) => new WalletConnectV2WithAnalytics({ actions, onError })
 )
-export const walletConnectV2Connection: Connection = {
-  getName: () => 'WalletConnectV2',
+export const walletConnectConnection: Connection = {
```
Comment (zzmp) [inline src/connection/index.ts:88]: I think this should remain named `walletConnectV2Connection` to help disambiguate, as long as both V1 and V2 exist in the codebase.

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/WalletModal/Option.tsx (hunk 2 above)
Comment (zzmp) [reply src/components/WalletModal/Option.tsx:185]: This is already answered in the PR description, thank you!

File: src/connection/index.ts (hunk 7 above)
Comment (zzmp) [reply src/connection/index.ts:88]: In general, I don't think anything should be named `WalletConnect` while both V1 and V2 exist in the codebase - they should all be suffixed.

File: src/connection/WalletConnectV2.ts (hunk 4 above)
Comment (JFrankfurt) [reply src/connection/WalletConnectV2.ts:35]: unfortunately there are no docs w/r/t this configuration option, but I'm happy to add a comment with what background info we do have

File: src/connection/WalletConnect.ts (hunk 3 above)
Comment (JFrankfurt) [reply src/connection/WalletConnect.ts:34]: https://linear.app/uniswap/issue/WEB-2346/update-uniswap-wallet-to-walletconnect-v2

https://github.com/Uniswap/interface/pull/6762

File: src/connection/index.ts (hunk 7 above)
Comment (JFrankfurt) [reply src/connection/index.ts:88]: that seems reasonable, but I'm leaving the one on line 89 because that's the string that gets rendered into the UI.

Comment (zzmp) [reply src/connection/index.ts:88]: yes agree, rendered UI is different than code

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by just-toby — APPROVED** [review]

**Pull request merged**